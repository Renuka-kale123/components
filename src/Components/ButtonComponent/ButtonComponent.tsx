import React from 'react';
import { TouchableOpacity, Text } from 'react-native';

  export const ButtonComponent = ({
    width,
    height,
    bgColor,
    title,
    titleColor,
    titleSize,
    borderRadius,
    accessibilityLabel,
    accessible,
    onPress,
  }) => {
    return (
  <TouchableOpacity
        style={{
          width,
          height,
          backgroundColor: bgColor,
          borderRadius,
          justifyContent: 'center',
          alignItems: 'center',
        }}
        accessibilityLabel={accessibilityLabel}
        accessible={accessible}
        onPress={onPress}
  >
  <Text style={{ color: titleColor, fontSize: titleSize }}>{title}</Text>
  </TouchableOpacity>
    );
  };