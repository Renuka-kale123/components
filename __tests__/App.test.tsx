/**
 * @format
 */



import 'react-native';
import React from 'react';
import ButtonComponent from '../Task/ButtonComponent';
import renderer from 'react-test-renderer';
import { fireEvent } from '@testing-library/react-native'; // Import fireEvent from @testing-library/react-native
 
// Test render Snapshot
test('Test render SnapShot', () => {
  const snapShot = renderer.create(
<ButtonComponent
      width={200}
      height={50}
      bgColor={'orange'}
      title={'my button'}
      titleColor={'black'}
      titleSize={20}
      borderRadius={10}
    />
  ).toJSON();
  expect(snapShot).toMatchSnapshot();
});
 
// Test click function
test('Test click function', () => {
  let clicked = false;
  const handleClick = () => {
    clicked = true;
  };
 
  const component = renderer.create(
<ButtonComponent
      width={200}
      height={50}
      bgColor={'orange'}
      title={'my button'}
      titleColor={'black'}
      titleSize={20}
      borderRadius={10}
      onPress={handleClick}
    />
  );
 
  const buttonInstance = component.root.findByType(ButtonComponent);
  fireEvent.press(buttonInstance); // Simulate a click event on the button
 
  expect(clicked).toBe(true); // Assert that the click function was called
});
 
// Test button styles
test('Test button styles', () => {
  const component = renderer.create(
<ButtonComponent
      width={200}
      height={50}
      bgColor={'orange'}
      title={'my button'}
      titleColor={'black'}
      titleSize={20}
      borderRadius={10}
    />
  );
  const buttonInstance = component.root.findByType(ButtonComponent);
 

  expect(buttonInstance.props.width).toBe(200);
  expect(buttonInstance.props.height).toBe(50);
  expect(buttonInstance.props.bgColor).toBe('orange');
  expect(buttonInstance.props.titleColor).toBe('black');
  expect(buttonInstance.props.titleSize).toBe(20);
  expect(buttonInstance.props.borderRadius).toBe(10);
});
 
// Test button accessibility
test('Test button accessibility', () => {
  const component = renderer.create(
<ButtonComponent
      width={200}
      height={50}
      bgColor={'orange'}
      title={'my button'}
      titleColor={'black'}
      titleSize={20}
      borderRadius={10}
      accessibilityLabel="My button" // Adding accessibility label
      accessible={true} // Making button accessible
    />
  );
  const buttonInstance = component.root.findByType(ButtonComponent);
 
  // Check button accessibility attributes
  expect(buttonInstance.props.accessibilityLabel).toBe('My button');
  expect(buttonInstance.props.accessible).toBe(true);
});
 
