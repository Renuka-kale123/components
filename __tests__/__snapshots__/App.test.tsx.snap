// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render SnapShot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "orange",
        "borderRadius": 10,
        "height": 50,
        "justifyContent": "center",
        "width": 200,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 20,
        }
      }
    >
      my button
    </Text>
  </View>
</View>
`;
